src/CodeGen_OpenCL_Dev.cpp

```diff
@@ -16,6 +17,131 @@ using std::sort;
 
 static ostringstream nil;
 
+// OpenCL doesn't support vectors of bools, this mutator rewrites IR
+// to use signed integer vectors instead.
+class EliminateBoolVectors : public IRMutator {
+private:
+    using IRMutator::visit;
+
+    template <typename T>
+    void visit_comparison(const T* op) {
+        Expr a = mutate(op->a);
+        Expr b = mutate(op->b);
+
+        Type t = a.type();
+        if (t.width > 1) {
+            // To represent bool vectors, OpenCL uses vectors of signed
+            // integers with the same width as the types being compared.
+            t.code = Type::Int;
+            expr = T::make(a, b);
```

Comment:
Good idea, done.

I'm not seeing why this code would fail for bool vector inputs... Do you mean if the arguments to the kernel are bool and vectorized in the kernel? OpenCL doesn't support that at all, so I don't think we need to worry about that case. Maybe an error if the user attempts to do that would be a good idea, but we have lots of other things like this that we aren't currently checking for either.